workflow/models.py

```diff
@@ -1386,10 +1386,27 @@ class Budget(models.Model):
     edit_date = models.DateTimeField(null=True, blank=True)
     created_by = models.ForeignKey('auth.User', related_name='budgets', null=True, blank=True)
     history = HistoricalRecords()
+
     # on save add create date or update edit date
     def save(self, *args, **kwargs):
-        if self.create_date == None:
+        if not self.create_date:
+            if self.workflowlevel2:
+                wflvl2 = self.workflowlevel2
+                wflvl2.total_estimated_budget += self.proposed_value
+                wflvl2.actual_cost += self.actual_value
+                wflvl2.save()
             self.create_date = timezone.now()
+        elif self.workflowlevel2:
+            wflvl2 = self.workflowlevel2
+            old_budget = self.__class__.objects.get(id=self.id)
```

Comment:
Would not it be better remove the redundant code inside the condition `if not self.create_date:` and place here an `if`?

And to get the old budget you can use a try, catch ObjectDoesNotExists.

Reply:
I got your point but it doesn't raise `ObjectDoesNotExists`, it raises `AttributeError`.

Reply:
Ok. you're talking about the `old_budget`. Sorry. hehe

Reply:
Done @Menda